openapi.yaml

```diff
@@ -25,105 +32,433 @@ security:
+          required: false
+          type: integer
+        - in: query
+          name: bakeTime
+          description: Search pizzas by baking time (in minutes)
+          type: integer
+        - in: query
+          name: ingredientsList
+          description: Search for pizzas with all the ingredients specified by this list of ingredient IDs
+          type: array
+          items:
+            type: string
+        - in: query
+          name: doughId
+          description: Search for pizzas with the dough matching this dough ID
+          type: string
       responses:
-        '200':
-          description: Successful response
+        "200":
+          description: A list of pizzas
           schema:
-            $ref: '#/definitions/PetsResult'
-        '400':
-          $ref: '#/responses/400'
-        '500':
-          $ref: '#/responses/500'
+            $ref: "#/definitions/PizzasResults"
+        "500":
+          $ref: "#/responses/500"
     post:
-      summary: Add a new pet
-      operationId: postPet
+      description: Create a new pizza recipe
       tags:
-        - pets
+        - pizza
       parameters:
         - in: body
-          name: body
+          name: attributes
+          description: attributes to assign to the new pizza
           required: true
           schema:
-            $ref: '#/definitions/PetPostBody'
+            $ref: "#/definitions/PizzaAttributes"
       responses:
-        '201':
-          description: Pet successfully created
+        "201":
+          description: The newly created pizza recipe
           schema:
-            $ref: '#/definitions/PetResult'
-        '400':
-          $ref: '#/responses/400'
-        '409':
-          description: Conflict
+            $ref: "#/definitions/PizzaResult"
+        "400":
+          $ref: "#/responses/400"
+        "500":
+          $ref: "#/responses/500"
+  /pizzas/{pizzaId}:
+    get:
+      description: Get a pizza with Id pizzaId
+      tags:
+        - pizza
+      parameters:
+        - $ref: "#/parameters/pizzaId"
+      responses:
+        "200":
+          description: A single pizza
           schema:
-            $ref: '#/definitions/ErrorResult'
-        '500':
-          $ref: '#/responses/500'
-  /pets/{id}:
+            $ref: "#/definitions/PizzaResult"
+        "404":
+          $ref: "#/responses/404"
+        "500":
+          $ref: "#/responses/500"
+    patch:
+      description: Update a pizza recipe
+      tags:
+        - pizza
+      parameters:
+        - $ref: "#/parameters/pizzaId"
+        - in: body
+          name: attributes
+          description: attributes of the pizza to be updated
+          schema:
+            $ref: "#/definitions/PizzaResource"
+      responses:
+        "200":
+          description: The updated pizza recipe
+          schema:
+            $ref: "#/definitions/PizzaResult"
+        "400":
+          $ref: "#/responses/400"
+        "404":
+          $ref: "#/responses/404"
```

Comment:
For PATCHing, a 409 is returned when type/id are invalid https://jsonapi.org/format/#crud-updating-responses-409